enumerable_methods.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    i = 0
+    result = []
+    while i < length
+      answer = yield(self[i])
+      result << self[i] unless answer == false
+      i += 1
+    end
+    result
+  end
+
+  def my_all?(classification = nil)
+    test = true
+    checked = classification.class
+    my_each { |item| return false unless yield[item] == false } if block_given?
+    my_each { |item| return false unless item.class == classification } if checked == Class
+    my_each { |item| return false unless item =~ classification } if checked == Regexp
+    my_each { |item| return false unless item == classification } if [Integer, String].include?(checked)
+    my_each { |item| return false unless item } if !classification && !block_given?
+    test
+  end
+
+  def my_any?(classification = nil)
+    test = false
+    my_each { |item| return true unless yield(item) == false } if block_given?
+    my_each { |item| return true unless item.class == classification } if classification.class == Class
+    my_each { |item| return true unless item =~ classification } if classification.class == Regexp
+    my_each { |item| return true unless item == classification } if [Integer, String].include?(classification.class)
+    my_each { |item| return true if item } if !classification && !block_given?
+    test
+  end
```

Comment:
**[REQUIRED]**
* When a class is passed as an argument returns true if at least one of the collection is a member of such class
```
Failure/Error: expect(array.my_any?(Integer)).to be array.any?(Integer)
     
       expected true
            got false
     # ./spec/enumerables_spec.rb:119:in `block (4 levels) in <top (required)>'
```
* When a Regex is passed as an argument returns false if none of the collection matches the Regex.
```
Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)
     
       expected false
            got true
     # ./spec/enumerables_spec.rb:126:in `block (4 levels) in <top (required)>'
```
* When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
     
       expected false
            got true
     # ./spec/enumerables_spec.rb:133:in `block (4 levels) in <top (required)>'
```